src/ripple/overlay/tests/short_read.test.cpp

```diff
@@ -533,14 +533,14 @@ class short_read_test : public beast::unit_test::suite
 
 public:
     short_read_test()
-        : work_(boost::in_place(std::ref(io_service_)))
-        , thread_(std::thread([this]()
+        : thread_(std::thread([this]()
             {
                 beast::Thread::setCurrentThreadName("io_service");
                 this->io_service_.run();
             }))
         , context_(make_SSLContext())
     {
+        work_.emplace (io_service_);
```

Comment:
LOL.. no. Did tests pass with this change? `io_service_.run()` will return immediately if `work_` is not constructed first. See: http://www.boost.org/doc/libs/1_58_0/doc/html/boost_asio/reference/io_service__work.html